### PR TITLE
Add support for queries in List attributes and List Indexes endpoint

### PR DESCRIPTION
**Pull request by fanatic75**
## What does this PR do?

Adds support for queries in the List Attributes endpoint and List Indexes endpoint

## Test Plan

Adds E2E test case to verify list attributes and list indexes with queries

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [x] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: app/controllers/api/databases.php [hunk 1]
```diff
@@ -1669,16 +1672,55 @@ function updateAttribute(
 
         $collection = $dbForProject->getDocument('database_' . $database->getInternalId(), $collectionId);
 
+
```
Comment (abnegate) [inline app/controllers/api/databases.php:1675]: Let's remove extra blank lines

File: app/controllers/api/databases.php [hunk 2]
```diff
@@ -1669,16 +1672,55 @@ function updateAttribute(
...
         if ($collection->isEmpty()) {
             throw new Exception(Exception::COLLECTION_NOT_FOUND);
         }
 
-        $attributes = $collection->getAttribute('attributes');
+        $queries = Query::parseQueries($queries);
+        foreach ($queries as $query) {
+            if ($query->getMethod() ===  Query::TYPE_SELECT) {
+                throw new Exception(Exception::GENERAL_QUERY_INVALID, "Select queries are not valid.");
```
Comment (abnegate) [inline app/controllers/api/databases.php:1683]: Let's use single quotes here

File: app/controllers/api/databases.php [hunk 3]
```diff
@@ -1669,16 +1672,55 @@ function updateAttribute(
...
-            'attributes' => $attributes
-        ]), Response::MODEL_ATTRIBUTE_LIST);
+         // Get cursor document if there was a cursor query
+         $cursor = Query::getByType($queries, Query::TYPE_CURSORAFTER, Query::TYPE_CURSORBEFORE);
+         $cursor = reset($cursor);
+
+        if ($cursor) {
+            $attributeId = $cursor->getValue();
+
+            $cursorDocument = Authorization::skip(fn() => $dbForProject->find('attributes', [Query::equal('collectionId', [$collectionId]), Query::equal('databaseId', [$databaseId]), Query::equal('key', [$attributeId]), Query::limit(1)]));
```
Comment (abnegate) [inline app/controllers/api/databases.php:1696]: Let's add a line breaks for readability:

```suggestion
            $cursorDocument = Authorization::skip(fn() => $dbForProject->find('attributes', [
                Query::equal('collectionId', [$collectionId]),
                Query::equal('databaseId', [$databaseId]),
                Query::equal('key', [$attributeId]), 
                Query::limit(1),
            ]));
```

File: app/controllers/api/databases.php [hunk 4]
```diff
@@ -1669,16 +1672,55 @@ function updateAttribute(
...
+                $options = $attribute->getAttribute('options');
+                $attribute->setAttribute('relatedCollection', $options['relatedCollection']);
+                $attribute->setAttribute('relationType', $options['relationType']);
+                $attribute->setAttribute('twoWay', $options['twoWay']);
+                $attribute->setAttribute('twoWayKey', $options['twoWayKey']);
+                $attribute->setAttribute('side', $options['side']);
+                $attribute->setAttribute('onDelete', $options['onDelete']);
+            }
+        }
+         $filterQueries = Query::groupByType($queries)['filters'];
```
Comment (abnegate) [inline app/controllers/api/databases.php:1719]: Looks like some indentation is off here

File: app/controllers/api/databases.php [hunk 5]
```diff
@@ -2480,26 +2522,50 @@ function updateAttribute(
...
 
         if ($collection->isEmpty()) {
             throw new Exception(Exception::COLLECTION_NOT_FOUND);
         }
 
-        $indexes = $collection->getAttribute('indexes');
+        $queries = Query::parseQueries($queries);
+        foreach ($queries as $query) {
+            if ($query->getMethod() ===  Query::TYPE_SEARCH) {
+                throw new Exception(Exception::GENERAL_QUERY_INVALID, "Select queries are not valid.");
```
Comment (abnegate) [inline app/controllers/api/databases.php:2545]: Let's use single quotes here

File: app/controllers/api/databases.php [hunk 6]
```diff
@@ -2480,26 +2522,50 @@ function updateAttribute(
...
+        \array_push($queries, Query::equal('collectionId', [$collectionId]), Query::equal('databaseId', [$databaseId]));
+
+         // Get cursor document if there was a cursor query
+         $cursor = Query::getByType($queries, Query::TYPE_CURSORAFTER, Query::TYPE_CURSORBEFORE);
+         $cursor = reset($cursor);
 
+        if ($cursor) {
+            $indexId = $cursor->getValue();
+
+            $cursorDocument = Authorization::skip(fn() => $dbForProject->find('indexes', [Query::equal('collectionId', [$collectionId]), Query::equal('databaseId', [$databaseId]), Query::equal('key', [$indexId]), Query::limit(1)]));
```
Comment (abnegate) [inline app/controllers/api/databases.php:2557]: Let's line break for readability

File: src/Appwrite/Utopia/Database/Validator/Queries/Attributes.php [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+namespace Appwrite\Utopia\Database\Validator\Queries;
+
+class Attributes extends Base
+{
+    public const ALLOWED_ATTRIBUTES = [
+        'key',
+        'type',
+        'size',
+        'required',
+        'array'
```
Comment (abnegate) [inline src/Appwrite/Utopia/Database/Validator/Queries/Attributes.php:12]: Let's add `status` and `error` here too

File: src/Appwrite/Utopia/Database/Validator/Queries/Indexes.php [hunk 8]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Appwrite\Utopia\Database\Validator\Queries;
+
+class Indexes extends Base
+{
+    public const ALLOWED_ATTRIBUTES = [
+        'key',
+        'type',
+        'status',
```
Comment (abnegate) [inline src/Appwrite/Utopia/Database/Validator/Queries/Indexes.php:10]: Let's add `error` here too

File: app/controllers/api/databases.php (hunk 1 above)
Comment (fanatic75) [reply app/controllers/api/databases.php]: done this

File: app/controllers/api/databases.php (hunk 2 above)
Comment (fanatic75) [reply app/controllers/api/databases.php]: done this

File: app/controllers/api/databases.php (hunk 3 above)
Comment (fanatic75) [reply app/controllers/api/databases.php]: done this

File: app/controllers/api/databases.php (hunk 4 above)
Comment (fanatic75) [reply app/controllers/api/databases.php]: done this

File: app/controllers/api/databases.php (hunk 5 above)
Comment (fanatic75) [reply app/controllers/api/databases.php]: done this

File: app/controllers/api/databases.php (hunk 6 above)
Comment (fanatic75) [reply app/controllers/api/databases.php]: done this

File: src/Appwrite/Utopia/Database/Validator/Queries/Attributes.php (hunk 7 above)
Comment (fanatic75) [reply src/Appwrite/Utopia/Database/Validator/Queries/Attributes.php]: added status, there is no error property in attributes

File: src/Appwrite/Utopia/Database/Validator/Queries/Indexes.php (hunk 8 above)
Comment (fanatic75) [reply src/Appwrite/Utopia/Database/Validator/Queries/Indexes.php:10]: didn't add it, there is no error property in db schema for indexes

**Comment by fanatic75** [comment]
@abnegate, can you please re-review again? I fixed all the lint issues, haven't added error as it's not found in Attributes or Indexes schema.

**Pull request closed without merging**